### PR TITLE
Update Heltec Automation(TM) boards defintion.

### PR DESCRIPTION
**Pull request by Heltec-Aaron-Lee**
Update Heltec Automation(TM) boards defintion, include WiFi_Kit_32, WiFi_LoRa_32, WiFi_LoRa_32_V2, Wireless_Stick defintions.

Please check.

### CODE REVIEW

**Comment by cyberman54** [comment]
> Update Heltec Automation(TM) boards defintion, include WiFi_Kit_32, WiFi_LoRa_32, WiFi_LoRa_32_V2, Wireless_Stick defintions.
> 
> Please check.

This will break compatibilty with Heltec v1 :-(

**Comment by stickbreaker** [comment]
@Heltec-Aaron-Lee I would recommend you consider the other menu choices for these boards.
```
{variant}.menu.PartitionScheme.default=Default
{variant}.menu.PartitionScheme.default.build.partitions=default
{variant}.menu.PartitionScheme.minimal=Minimal (2MB FLASH)
{variant}.menu.PartitionScheme.minimal.build.partitions=minimal
{variant}.menu.PartitionScheme.no_ota=No OTA (Large APP)
{variant}.menu.PartitionScheme.no_ota.build.partitions=no_ota
{variant}.menu.PartitionScheme.no_ota.upload.maximum_size=2097152
{variant}.menu.PartitionScheme.huge_app=Huge APP (3MB No OTA)
{variant}.menu.PartitionScheme.huge_app.build.partitions=huge_app
{variant}.menu.PartitionScheme.huge_app.upload.maximum_size=3145728
{variant}.menu.PartitionScheme.min_spiffs=Minimal SPIFFS (Large APPS with OTA)
{variant}.menu.PartitionScheme.min_spiffs.build.partitions=min_spiffs
{variant}.menu.PartitionScheme.min_spiffs.upload.maximum_size=1966080
{variant}.menu.PartitionScheme.fatflash=16M Fat
{variant}.menu.PartitionScheme.fatflash.build.partitions=ffat

{variant}.menu.CPUFreq.240=240MHz (WiFi/BT)
{variant}.menu.CPUFreq.240.build.f_cpu=240000000L
{variant}.menu.CPUFreq.160=160MHz (WiFi/BT)
{variant}.menu.CPUFreq.160.build.f_cpu=160000000L
{variant}.menu.CPUFreq.80=80MHz (WiFi/BT)
{variant}.menu.CPUFreq.80.build.f_cpu=80000000L
{variant}.menu.CPUFreq.40=40MHz (40MHz XTAL)
{variant}.menu.CPUFreq.40.build.f_cpu=40000000L
{variant}.menu.CPUFreq.26=26MHz (26MHz XTAL)
{variant}.menu.CPUFreq.26.build.f_cpu=26000000L
{variant}.menu.CPUFreq.20=20MHz (40MHz XTAL)
{variant}.menu.CPUFreq.20.build.f_cpu=20000000L
{variant}.menu.CPUFreq.13=13MHz (26MHz XTAL)
{variant}.menu.CPUFreq.13.build.f_cpu=13000000L
{variant}.menu.CPUFreq.10=10MHz (40MHz XTAL)
{variant}.menu.CPUFreq.10.build.f_cpu=10000000L

{variant}.menu.FlashSize.4M=4MB (32Mb)
{variant}.menu.FlashSize.4M.build.flash_size=4MB
{variant}.menu.FlashSize.2M=2MB (16Mb)
{variant}.menu.FlashSize.2M.build.flash_size=2MB
{variant}.menu.FlashSize.2M.build.partitions=minimal
{variant}.menu.FlashSize.16M=16MB (128Mb)
{variant}.menu.FlashSize.16M.build.flash_size=16MB
{variant}.menu.FlashSize.16M.build.partitions=ffat

{variant}.menu.UploadSpeed.921600=921600
{variant}.menu.UploadSpeed.921600.upload.speed=921600
{variant}.menu.UploadSpeed.115200=115200
{variant}.menu.UploadSpeed.115200.upload.speed=115200
{variant}.menu.UploadSpeed.256000.windows=256000
{variant}.menu.UploadSpeed.256000.upload.speed=256000
{variant}.menu.UploadSpeed.230400.windows.upload.speed=256000
{variant}.menu.UploadSpeed.230400=230400
{variant}.menu.UploadSpeed.230400.upload.speed=230400
{variant}.menu.UploadSpeed.460800.linux=460800
{variant}.menu.UploadSpeed.460800.macosx=460800
{variant}.menu.UploadSpeed.460800.upload.speed=460800
{variant}.menu.UploadSpeed.512000.windows=512000
{variant}.menu.UploadSpeed.512000.upload.speed=512000

{variant}.menu.DebugLevel.none=None
{variant}.menu.DebugLevel.none.build.code_debug=0
{variant}.menu.DebugLevel.error=Error
{variant}.menu.DebugLevel.error.build.code_debug=1
{variant}.menu.DebugLevel.warn=Warn
{variant}.menu.DebugLevel.warn.build.code_debug=2
{variant}.menu.DebugLevel.info=Info
{variant}.menu.DebugLevel.info.build.code_debug=3
{variant}.menu.DebugLevel.debug=Debug
{variant}.menu.DebugLevel.debug.build.code_debug=4
{variant}.menu.DebugLevel.verbose=Verbose
{variant}.menu.DebugLevel.verbose.build.code_debug=5
```

**Comment by Heltec-Aaron-Lee** [comment]
@stickbreaker should I undo this merge and undo again? or I can submission a new pull request it in the future update?

**Comment by stickbreaker** [comment]
@heltec-aaron-lee Just modify the effected files.  The pr points to a specific branch. Just modify that branch.

**Comment by me-no-dev** [comment]
I guess I will wait for the additions?

**Comment by n9wxu** [comment]
Please consider the modifications to the partition map for the board support.  The board has a larger flash than the default 4MB.  I included an 8MB partition map in pr #2407

**Comment by n9wxu** [comment]
That was for the LoRa32 V2 hardware.

**Comment by Heltec-Aaron-Lee** [comment]
@me-no-dev @n9wxu @stickbreaker I had submission a new push request [#2470](https://github.com/espressif/arduino-esp32/pull/2470).

I think this version is fine :D

**Comment by me-no-dev** [comment]
@Heltec-Aaron-Lee thanks! will close this one now :)

@n9wxu does the new PR cover your requirements?

File: boards.txt [hunk 1]
```diff
@@ -2119,6 +2119,94 @@ heltec_wifi_lora_32.menu.UploadSpeed.512000.upload.speed=512000
...
+heltec_wifi_lora_32_V2.build.mcu=esp32
+heltec_wifi_lora_32_V2.build.core=esp32
+heltec_wifi_lora_32_V2.build.variant=heltec_wifi_lora_32_V2
+heltec_wifi_lora_32_V2.build.board=Heltec_WIFI_LoRa_32_V2
+
+heltec_wifi_lora_32_V2.build.f_cpu=240000000L
+heltec_wifi_lora_32_V2.build.flash_mode=dio
+heltec_wifi_lora_32_V2.build.flash_size=4MB
+heltec_wifi_lora_32_V2.build.boot=dio
+heltec_wifi_lora_32_V2.build.partitions=default
```
Comment (n9wxu) [inline boards.txt]: The default LoRa32 v2 has 8MB of flash and needs a different partition map.  This can be found in PR #2407

File: boards.txt [hunk 2]
```diff
@@ -2119,6 +2119,94 @@ heltec_wifi_lora_32.menu.UploadSpeed.512000.upload.speed=512000
...
+heltec_wifi_lora_32_V2.serial.disableRTS=true
+
+heltec_wifi_lora_32_V2.build.mcu=esp32
+heltec_wifi_lora_32_V2.build.core=esp32
+heltec_wifi_lora_32_V2.build.variant=heltec_wifi_lora_32_V2
+heltec_wifi_lora_32_V2.build.board=Heltec_WIFI_LoRa_32_V2
+
+heltec_wifi_lora_32_V2.build.f_cpu=240000000L
+heltec_wifi_lora_32_V2.build.flash_mode=dio
+heltec_wifi_lora_32_V2.build.flash_size=4MB
```
Comment (n9wxu) [inline boards.txt]: This should be 8MB to cover the flash on the V2 HW.

Comment (Heltec-Aaron-Lee) [reply boards.txt]: Thanks for your notice @n9wxu.

@me-no-dev I had submisson a new PR [#2474](https://github.com/espressif/arduino-esp32/pull/2474), it's include a new partition map as @n9wxu suggested. This one is OK and I will close PR [#2470](https://github.com/espressif/arduino-esp32/pull/2470)

I think [#2407](https://github.com/espressif/arduino-esp32/pull/2407) can be closed too.

Comment (n9wxu) [reply boards.txt]: Looks good.

**Pull request closed without merging**